pipeline/stage02_processing/scripts/spatial_downsampling.py

```diff
@@ -16,9 +16,29 @@
 def spatial_smoothing(images, MACRO_PIXEL_DIM):
 
     # Now we need to reduce the noise from the images by performing a spatial smoothing
-    images = measure.block_reduce(images, (1, MACRO_PIXEL_DIM, MACRO_PIXEL_DIM), np.mean, cval = np.median(images))
+    images_reduced = measure.block_reduce(images, (1, MACRO_PIXEL_DIM, MACRO_PIXEL_DIM), np.mean, cval = np.median(images))
 
-    return images
+    if args.output_img is not None:
+        plt.figure()
+        plt.imshow(images_reduced[0], interpolation='nearest', cmap='viridis', origin='lower')
+        save_plot(args.output_img)
```

Comment:
Each function should ideally perform only one task and use only its explicitly defined arguments.
Thus, the plotting should happen separate from the smoothing function.